catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -81,9 +85,99 @@
     private static final FastDateFormat DATE_FORMAT = FastDateFormat.getInstance(
             "yyyy-MM-dd'T'HH:mm:ssZZ");
 
+    private final CheckedFunction<InputStream, PDDocument> inputStreamPDDocumentFunction;
+
+    private final CheckedFunction<PDDocument, String> geoParserFunction;
+
+    private final CheckedFunction<PDDocument, byte[]> thumbnailFunction;
+
     private Map<ServiceReference, ContentMetadataExtractor> contentMetadataExtractors =
             Collections.synchronizedMap(new TreeMap<>(new ServiceComparator()));
 
+    private MetacardType metacardType;
+
+    private boolean usePdfTitleAsTitle;
+
+    /**
+     * @param metacardType       must be non-null
+     * @param usePdfTitleAsTitle must be non-null
+     */
+    public PdfInputTransformer(MetacardType metacardType, Boolean usePdfTitleAsTitle) {
+        this(metacardType,
+                usePdfTitleAsTitle,
+                PDDocument::load,
+                GEO_PDF_PARSER::getWktFromPDF,
+                PdfInputTransformer::generatePdfThumbnail);
+    }
+
+    /**
+     * Note: direct use of this constructor is meant for unit testing only
+     *
+     * @param metacardType                  must be non-null
+     * @param usePdfTitleAsTitle            must be non-null
+     * @param inputStreamPDDocumentFunction must be non-null
+     * @param geoParserFunction             must be non-null
+     * @param thumbnailFunction             must be non-null
+     */
+    PdfInputTransformer(MetacardType metacardType, Boolean usePdfTitleAsTitle,
+            CheckedFunction<InputStream, PDDocument> inputStreamPDDocumentFunction,
+            CheckedFunction<PDDocument, String> geoParserFunction,
+            CheckedFunction<PDDocument, byte[]> thumbnailFunction) {
+        notNull(metacardType, "metacardType must be non-null");
+        notNull(usePdfTitleAsTitle, "usePdfTitleAsTitle must be non-null");
+        notNull(inputStreamPDDocumentFunction, "inputStreamPDDocumentFunction must be non-null");
+        notNull(geoParserFunction, "geoParserFunction must be non-null");
+        notNull(thumbnailFunction, "thumbnailFunction must be non-null");
+
+        this.metacardType = metacardType;
+        this.usePdfTitleAsTitle = usePdfTitleAsTitle;
+        this.inputStreamPDDocumentFunction = inputStreamPDDocumentFunction;
+        this.geoParserFunction = geoParserFunction;
+        this.thumbnailFunction = thumbnailFunction;
+    }
+
+    @SuppressWarnings("StatementWithEmptyBody")
+    private static byte[] generatePdfThumbnail(PDDocument pdfDocument) throws IOException {
+
+        PDFRenderer pdfRenderer = new PDFRenderer(pdfDocument);
+
+        if (pdfDocument.getNumberOfPages() < 1) {
+            /*
+             * Can there be a PDF with zero pages??? Should we throw an error or what? The
```

Comment:
If the PDF has zero pages I don't think we need to worry about thumbnail generation.